### PR TITLE
Add support for strict webhook failure policy

### PR DESCRIPTION
**Pull request by khogeland**
Allows users to enable a strict failure policy in the webhook config. Requires that a namespace selector be set for the webhook, as an operator pod that is malfunctioning or a lack of healthy operator instances could cause a cluster-wide outage (and prevent the operator itself from being rescheduled).

### CODE REVIEW

File: pkg/webhook/webhook.go [hunk 1]
```diff
@@ -122,10 +143,12 @@ func (wh *WebHook) Start(webhookConfigName string) error {
 
 // Stop deregisters itself with the API server and stops the admission webhook server.
 func (wh *WebHook) Stop(webhookConfigName string) error {
-	if err := wh.selfDeregistration(webhookConfigName); err != nil {
-		return err
+	if wh.failurePolicy != arv1beta1.Fail {
```
Comment (liyinan926) [inline pkg/webhook/webhook.go]: Can you explain why this change is needed?

File: pkg/webhook/webhook.go [hunk 2]
```diff
@@ -91,6 +99,19 @@ func New(
 		cert:              cert,
 		serviceRef:        serviceRef,
 		sparkJobNamespace: jobNamespace,
+		failurePolicy:     failurePolicy,
+	}
+
+	if webhookNamespaceSelector != "" {
+		kv := strings.SplitN(webhookNamespaceSelector, "=", 2)
```
Comment (liyinan926) [inline pkg/webhook/webhook.go]: Should we allow more multi-selector, e.g., `key1=val1,key2=val2`?

File: pkg/webhook/webhook.go (hunk 1 above)
Comment (khogeland) [reply pkg/webhook/webhook.go]: I'll add a comment. If the webhook deregisters itself, some pods may be scheduled without hitting the webhook (for example, if the webhook gets deregistered while a driver is creating executors).

File: pkg/webhook/webhook.go (hunk 2 above)
Comment (khogeland) [reply pkg/webhook/webhook.go]: I think it's best to keep this as simple until given a reason otherwise. If filtering on multiple values is needed, users can use e.g. `foo=sparkoperatorv123` instead of `sparkoperator=yes,sparkoperatorversion=123`.
